test/e2e/tests/request-queuing/ui.spec.js

```diff
@@ -61,7 +61,7 @@ async function openDappAndSwitchChain(driver, dappUrl, chainId) {
           (permission) =>
             permission.parentCapability === PermissionNames.permittedChains,
         )
-        ?.caveats.find(
+        ?.caveats?.find(
```

Comment:
When would `caveats` be missing here?

Reply:
Right, but the change on this line only impacts the scenario where `permittedChains` is found, but it's missing `caveats`.

Reply:
```suggestion
        ?.caveats.find(
```